src/test/browser/things-view/thing-test.js

```diff
@@ -1,11 +1,78 @@
 const {getBrowser} = require('../browser-common');
-const {addThing, saveStepScreen} = require('./test-utils');
+const {
+  addThing,
+  getProperty,
+  waitForExpect,
+} = require('./test-utils');
+const ThingsPage = require('../page-object/things-page');
 
 const STATIC_JS_PATH = '../../../../static/js';
 const Utils = require(`${STATIC_JS_PATH}/utils`);
 
 
 describe('Thing', () => {
+  it('should render a thing and be able to change properties', async () => {
+    const browser = getBrowser();
+    const desc = {
+      id: 'UnknownThings',
+      name: 'foofoo',
+      type: 'thing',
+      properties: {
+        numberProp: {
+          value: 10,
+          type: 'number',
+          unit: 'percent',
+        },
+        stringProp: {
+          value: 'bar',
+          type: 'string',
+        },
+        booleanProp: {
+          value: true,
+          type: 'boolean',
+        },
+      },
+    };
+    await addThing(desc);
+
+    const thingsPage = new ThingsPage(browser);
+    thingsPage.open();
+
+    await thingsPage.waitForThings();
+    const things = await thingsPage.things();
+    expect(things.length).toEqual(1);
+    const thingName = await things[0].thingName();
+    expect(thingName).toEqual('foofoo');
+
+    const detailPage = await things[0].openDetailPage();
+
+    await detailPage.waitForBooleanProperties();
+    const booleanProps = await detailPage.booleanProperties();
+    expect(booleanProps.length).toEqual(1);
+    const booleanValue = await booleanProps[0].getValue();
+    expect(booleanValue).toEqual('on');
+
+    const numberProps = await detailPage.numberProperties();
+    expect(numberProps.length).toEqual(1);
+    let numberValue = await numberProps[0].getValue();
+    expect(numberValue).toEqual('10');
```

Comment:
Why is numberValue a string here, but a number on line 62?

Reply:
`getProperty()` calls `get /things/:thingsId/:propertyName`, and the Gateway returns numberProperty as Number.
`getValue()` call the WebdriverIO API which get value from input field https://github.com/mozilla-iot/gateway/pull/983/files#diff-010cb46695bc359934ef4b599d5221e0R20. The WebdriverIO returns String value if input is Number.

Since this behavior is difficult to understand by code, I will cast a value if numberProperties.